src/main/scala/com/linkedin/feathr/offline/anchored/keyExtractor/MVELSourceKeyExtractor.scala

```diff
@@ -86,5 +86,6 @@ private[feathr] class MVELSourceKeyExtractor(val anchorExtractorV1: AnchorExtrac
   // this helps to reduce the number of joins
   // to the observation data
   // The default toString does not work, because toString of each object have different values
```

Comment:
Side comment: This is beyond this scope of this change, but it looks dubious to be using .toString to drive behavior like this. There should have been another function added, to drive the behavior described here. (Is this the reason you needed to add something to the .toString implementation below ...?)